cmake/compiler/compiler_flags_template.cmake

```diff
@@ -139,3 +139,7 @@ set_compiler_property(PROPERTY warning_shadow_variables)
 # Compiler flags to avoid recognizing built-in functions
 set_compiler_property(PROPERTY no_builtin)
 set_compiler_property(PROPERTY no_builtin_malloc)
+
+# Compiler flag for defining specs. Used only by gcc, other compilers may keep
```

Comment:
Should we eventually plan on supporting clang's `--config` using this same mechanism?

Reply:
I was thinking of that, and some clang/LLVM releases, like LLVM for Arm, does provide dedicated configs for this, but i'm unsure how well established / standardized this principle is, so for a start it's kept a bit independent in `cmake/compiler/clang/target.cmake`.
and also because it's needed for the existing compile check functionality.